src/api/app/services/scm_status_reporter.rb

```diff
@@ -24,7 +24,15 @@ def call
                                          @state,
                                          status_options)
     end
-  rescue Octokit::Error, Octokit::InvalidRepository, Gitlab::Error::Error => e
+  rescue Octokit::InvalidRepository
+    package = Package.find_by_project_and_name(@event_payload[:project], @event_payload[:package])
+    return if package.blank?
+
+    token = Token::Workflow.find_by(scm_token: @scm_token)
```

Comment:
A user can create multiple tokens with the same *scm_token*. We need to handle this. I haven't tested the following code, but something like that should be fine:

```ruby
    tokens = Token::Workflow.where(scm_token: @scm_token).pluck(:id)
    return if tokens.none?

    EventSubscription.where(channel: 'scm', token: tokens, package: package).delete_all
```

Reply:
nice!